e_methods.rb

```diff
@@ -0,0 +1,169 @@
+    elsif !argum.nil? && (argum.is_a? Class)
+      puts 'warning: given block not used'
+      my_each { |element| return true if element.is_a? argum }
+    elsif block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true if element == argum }
+    elsif block_given?
+      my_each { |element| return true if yield(element) }
+    elsif argum.nil?
+      my_each { |element| return true if element.nil? || element == false }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true if argum.match(element) }
+    elsif argum
+      my_each { |element| return true if element == argum }
+    else
+      my_each { |element| return true if element }
+    end
+    false
+  end
+
+  def my_none?(argum = nil)
+    if block_given? && !argum.nil?
+      puts 'warning: given block not used'
+      my_each { |element| return true unless element.is_a? argum }
+    elsif is_a?(Range) && block_given?
+      to_a.my_each { |element| return true if yield(element) == false }
+    elsif block_given?
+      to_a.my_each { |element| return true unless yield(element) == true }
+    elsif argum.nil?
+      my_each { |element| return false if element == true }
+      my_each { |element| return true if element == false }
+    elsif !argum.nil? && (argum.is_a? Class)
+      my_each { |element| return true unless element.is_a? argum }
+    elsif !argum.nil? && argum.instance_of?(Regexp)
+      my_each { |element| return true unless argum.match(element) }
+    elsif argum
+      my_each { |element| return false if element == argum }
+      my_each { |element| return true if element != argum }
+    else
+      my_each { |element| return true unless element }
+    end
+    false
+  end
+
+  def my_count(argum = nil)
+    counter = 0
+    if block_given?
+      my_each { |element| counter += 1 if yield(element) == true }
+    elsif argum
+      my_each { |element| counter += 1 if element == argum }
+    else
+      my_each { |element| counter += 1 if element }
+    end
+    counter
+  end
+
+  def my_map(prc = nil)
+    return to_enum(:my_map) unless block_given? || !prc.nil?
+
+    new_arr = []
+    if !prc.nil?
+      to_a.my_each { |i| new_arr.push(prc.call(i)) }
+    else
+      to_a.my_each { |i| new_arr.push(yield(i)) }
+    end
+    new_arr
+  end
+
+  def my_inject(argum = nil, sym = nil)
+    return 'LocalJumpError (block not given)' if !block_given? && !argum
+
+    if !block_given?
+      if argum && sym
+        memo = argum
+        to_a.my_each { |element| memo = memo.send(sym, element) }
+      elsif argum
+        memo = first
+        to_a[1..to_a.length].my_each { |element| memo = memo.send(argum, element) }
+      end
+    elsif block_given? && argum
+      memo = argum
+      my_each { |element| memo = yield(memo, element) }
+    else
+      memo = first
+      to_a[1..to_a.length].my_each { |element| memo = yield(memo, element) }
+      memo
+    end
+    memo
+  end
```

Comment:
and this [review](https://github.com/carloso0114/ruby_enumerators/pull/1/files#r536390674) you have forgotten to work on, it's super clear I think.

`
Should raise a "LocalJumpError" when no block or argument is given
Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
expected LocalJumpError but nothing was raised
`